src/sentry/utils/snuba.py

```diff
@@ -1023,6 +1010,11 @@ def constrain_condition_to_dataset(cond, dataset):
     """
     index = get_function_index(cond)
     if index is not None:
+        # IN conditions are detected as a function but aren't really.
+        if cond[index] == "IN":
```

Comment:
Is this related to alias resolution?

Reply:
Yes. `environment` conditions used to be pre-parsed and aliased by the event_search parser and would not have their columns re-aliased because of `skip_conditions`. Now, because all conditions are re-aliased the `IN` string is followed by a list which quacks like a function `get_function_index`.

Reply:
For posterity, I've added some tests to surface this issue in https://github.com/getsentry/sentry/pull/15214

Reply:
Do we need to fix `get_function_index` then? IIRC "IN" condition shouldn't quack like a `get_function_index` since function expression should always have a list/tuple in the second position and where IN just has the string "IN"

Reply:
I think the answer is yes. 

I think @markstory mentioned to me that this may be resolved in this PR, or in an independent PR.

Reply:
I can do a follow up pull request to fix `IN` being treated as a function.